build/build.go

```diff
@@ -594,9 +605,28 @@ func (db *DeploymentBuilder) buildPlatformNetworks(d *Deployment) error {
 				break
 			}
 		}
+		for _, nap := range networkAddrPools {
+			for _, n := range results {
+				if n.Name == nap.NetworkName {
+					pools_to_be_exported = append(pools_to_be_exported, p.Name)
+					associated_pools = append(associated_pools, p.Name)
+				}
+			}
+		}
+		for _, pool_name := range pools_to_be_exported {
+			for _, pool := range pools {
+				if pool.Name == pool_name {
+					addrpool, err = starlingxv1.NewAddressPool(db.namespace, pool)
+					if err != nil {
+						return err
+					}
+					d.AddressPools = append(d.AddressPools, addrpool)
+				}
+			}
+		}
```

Comment:
Can we do the following:
1. Let's write a separate function "GetAssociatedAddressPools" which takes network_name and []NetworkAddressPool and returns []AddressPool.
2. Append AddressPool.Name for associated assoiated_pools.
3. Use []AddressPool to create NewAddressPool and append to d.AddressPools which will be more straightforward.

I feel that would make it much more readable.

Please include UT for this function as well.

Reply:
Changes are done